### PR TITLE
Migrate hard coded images to Maven properties

### PR DESCRIPTION
**Pull request by gtroitsk**
### Summary

- Migrate all hardcoded images with versions into BOM
- Properties are now resolved by using ImageUtil from FW

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Dependency update
- [x] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by rsvoboda** [comment]
@michalvavrik could you take a look at this?

**Comment by michalvavrik** [comment]
@rsvoboda I talked to @gtroitsk and suggested what he should do in FW first, before this (basically we agreed on moving utils class there as it becomes repeated pattern, please note there are other tests in TS already that does already same thing). now @gtroitsk need to make changes when he has assigned ticket

**Comment by gtroitsk** [comment]
run tests

**Comment by michalvavrik** [comment]
@gtroitsk 9 commits for this PR is bit excessive, please squash them

File: pom.xml [hunk 1]
```diff
@@ -276,6 +281,7 @@
                                 <infinispan.image>quay.io/infinispan/server:14.0</infinispan.image>
                                 <infinispan.expected-log>Infinispan Server.*started in</infinispan.expected-log>
                                 <spring.cloud.server.image>quay.io/quarkusqeteam/spring-cloud-config-server:3.0</spring.cloud.server.image>
+                                <keycloak.image>quay.io/keycloak/keycloak:22.0.5</keycloak.image>
```
Comment (michalvavrik) [inline pom.xml:284]: I'm afraid you should wait for https://github.com/quarkus-qe/quarkus-test-suite/pull/1581 to get in and then rebase, not sure whether 22.0.5 is going to be needed.

File: messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/DevModeRedPandaDevServiceUserExperienceIT.java [hunk 2]
```diff
@@ -15,18 +18,18 @@
...
     static RestService app = new RestService()
             .withProperty("quarkus.kafka.devservices.enabled", Boolean.TRUE.toString())
-            .withProperty("quarkus.kafka.devservices.image-name", String.format("%s:%s", RED_PANDA_IMAGE, RED_PANDA_VERSION))
+            .withProperty("quarkus.kafka.devservices.image-name", "${redpanda.image}")
             .onPreStart(s -> DockerUtils.removeImage(RED_PANDA_IMAGE, RED_PANDA_VERSION));
 
     @Test
     public void verifyIfUserIsInformedAboutRedPandaDevServicePulling() {
-        app.logs().assertContains("Pulling docker image: vectorized/redpanda");
+        app.logs().assertContains("Pulling docker image: redpandadata/redpanda");
```
Comment (michalvavrik) [inline messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/DevModeRedPandaDevServiceUserExperienceIT.java:32]: I suppose this is fine, but can you provide me with context why this changed? Is it just renamed product as version have changed?

Comment (gtroitsk) [reply messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/DevModeRedPandaDevServiceUserExperienceIT.java:32]: vectorized/redpanda repo was deprecated and was migrated to redpandadata/redpanda

Comment (michalvavrik) [reply messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/DevModeRedPandaDevServiceUserExperienceIT.java:32]: Thank you

**Comment by gtroitsk** [comment]
run tests

**Comment by gtroitsk** [comment]
I think OC fails are unrelated

**Review by michalvavrik — APPROVED** [review]

**Pull request merged**